### ISSUE TITLE
TS Error when crafting a Transaction: Expression produces a union type that is too complex to represent.

### ISSUE THREAD

**Issue by Tbaut** [polkadot-api/polkadot-api#681]
Although this is known to you, I figured I'd create an issue so that I can track its resolution.

When crafting a Tx, with a dynamically typed api, TS may complain.
Here's an example:
```ts

interface Params {
  api: ApiType // this is of type TypedApi<typeof dot> | TypedApi<typeof ksm>
  threshold: number
  otherSignatories: string[]
  tx: Transaction<any, any, any, any>
  weight?: { ref_time: bigint; proof_size: bigint }
  when?: MultisigStorageInfo['when']
}

export const getAsMultiTx = ({
  api,
  threshold,
  otherSignatories,
  tx,
  weight,
  when
}: Params): Transaction<any, any, any, any> | undefined => {

  return api.tx.Multisig.as_multi({
    threshold,
    other_signatories: otherSignatories,
    maybe_timepoint: when,
    max_weight: weight || { proof_size: 0n, ref_time: 0n },
    call: tx.decodedCall
  })
}
```

The TS error "Expression produces a union type that is too complex to represent." comes from the fact that the `call` param in the `mulitig.as_multi` is strongly typed, and contains in my case the union of call types from polkadot, kusama and potentially all the other typed api that my application supports. Since all chains have different pallets, there's no overlap.

**Comment by Tbaut** [comment]
Thanks Josep, also good to note for anyone seeing this, it's not a blocking issue since there's an easy workaround for now, by narrowing the type on the api: `(api as TypedApi<typeof dot>).tx.Multisig...`

**Comment by carlosala** [comment]
This should be fixed after #687 🎉
It has been released at `polkadot-api@1.2.0`. We'll update the docs at papi.how soon!

**Comment by Tbaut** [comment]
Whoop whoop, so fast!! Thank you so much :pray:
